router/router.go

```diff
@@ -247,3 +248,27 @@ func (fs *onlyImageFS) Open(name string) (http.File, error) {
 	}
 	return fs.inner.Open(name)
 }
+
+func useXRealIP(trustedProxy bool, inner gin.LogFormatter) gin.LogFormatter {
+	return func(params gin.LogFormatterParams) string {
+		params.ClientIP = getClientIP(trustedProxy, params.Request)
+		return inner(params)
+	}
+}
+
+func getClientIP(trustedProxy bool, req *http.Request) string {
+	if trustedProxy {
+		realIPParts := strings.SplitN(req.Header.Get("x-real-ip"), ",", 2)
```

Comment:
Hmm, it seems like most applications use x-forwarded-for. I'm unsure again if this is a better solution or x-forwarded-for should be used instead. Let me think a little about this .____.

Reply:
Sorry to change my opinion this many times. X-real-ip seems to be legacy so x-forwarded for seems to be the standard. I changed this back to your original implementation as 
I don't really think it's worth to custom implement this checks for only ip logging, so your workaround with setting the ip to 127.0.0.1 when the connection originates from a socket is good enough.

I've simplified the socket check to `Request.RemoteAddr == "@"`, added the setting to the example config file and now always all SetTrustedProxies even if the list is empty. This is to prevent the bug described in https://github.com/gotify/server/issues/554#issuecomment-1560089638.